org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/IClassContent.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.core.compiler;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+
+/**
+ * Encapsulates the produced class of the compile operation
+ * @since 3.33
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+public interface IClassContent {
+
+	/**
+	 * @return the class bytes of this class content or <code>null</code> if no content was written
+	 * @throws CoreException if gathering the bytes failed
+	 */
+	byte[] getBytes() throws CoreException;
+
+	/**
+	 * Set the given classbytes to the provided array
+	 * @param classBytes
+	 * @throws CoreException
+	 */
+	void setBytes(byte[] classBytes) throws CoreException;
```

Comment:
Same as getBytes(), name similar to writeBytes() makes it apparent.
Having said that, who, other than the compiler is going to be using this method? What's the need to expose this to the outside world? If someone gets hold of an IClassContent and invoke this method, what is supposed to happen?

Reply:
See above, the bytes are not necessarily written as a consequence of this call (they might finally end up on disk even though).
The idea for this method is that for example a class-weaving function can get the bytes, do some weaving and later on update the bytes for further processing by other participants.